nml/actions/action2.py

```diff
@@ -101,9 +101,10 @@ def prepare_output(self, sprite_num):
         except IndexError:
             raise generic.ScriptError("Unable to allocate ID for [random]switch, sprite set/layout/group or produce-block. Try reducing the number of such blocks.", self.pos)
 
-    def write_sprite_start(self, file, size):
+    def write_sprite_start(self, file, size, extra_comment = []):
```

Comment:
[list as default argument has issues](https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments)

Reply:
Hmm, I did it like it's done in other locations. I guess all the code needs to be checked for this issue.